rootfs/api/models/build.py

```diff
@@ -52,8 +52,8 @@ def version(self):
         return 'git-{}'.format(self.sha) if self.source_based else 'latest'
 
     def create(self, user, *args, **kwargs):
-        latest_release = self.app.release_set.latest()
-
+        latest_release = self.app.release_set.filter(failed=False).latest()
```

Comment:
Yeah, big +1 on that as a general rule of thumb - I had more context on this since @kmala and I had talked about this at length a while back